src/platform/tls_schannel.c

```diff
@@ -1249,62 +1260,27 @@ CxPlatTlsSecConfigCreate(
         goto Error;
     }
 
+    CXPLAT_DBG_ASSERT(AchContext->SecConfig != NULL);
+    AchContext->SecConfig->ImpersonationToken =
+        PsReferenceImpersonationToken(
+            PsGetCurrentThread(),
+            &AchContext->SecConfig->CopyOnOpen,
+            &AchContext->SecConfig->EffectiveOnly,
+            &AchContext->SecConfig->ImpersonationLevel);
+
+    if (AchContext->SecConfig->ImpersonationToken == NULL) {
+        AchContext->SecConfig->ImpersonationToken =
+            PsReferencePrimaryToken(
+                PsGetCurrentProcess());
+        AchContext->SecConfig->IsPrimaryToken = TRUE;
+    }
```

Comment:
should assert here that the impersonation token is non-null.

Reply:
The code handles NULL, so I think we're fine without. I'm not sure if there is a scenario where it's supposed to be NULL.